hax/hax/queue/__init__.py

```diff
@@ -96,3 +100,22 @@ def to_ha_state(self, objinfo: Dict[str, str]) -> Optional[HAState]:
             LOG.error('Invalid json payload, no key (%s) present', error)
             return None
         return HAState(sdev_fid, status=state)
+
+    def handle_ha_notify(self, payload: Dict[str, Any]) -> None:
+        hastates: List[HAState] = []
+        for state in payload:
+            data = json.loads(json.dumps(state))
```

Comment:
Why do you need to re-encode it? Is it a workaround for some problem?
In other words, why can't we just say
```python
data = state
```

Reply:
The payload in this case is a list of dictionary strings, i.e. [{fid:0x000.., status: FAILED}], while trying to iterate over the list and indexing in a dictionary, was getting an error as `Invalid index type "str" for "str" for Union[int, slice]` . After loading the individual dictionary string I was able to index into the dictionary.

Reply:
Then probably it is safe to change the type hint of payload?

BTW mypy is very dumb, so there is always a possibility to force the type like this:
myvalue: str = mydict['key']
or myvalue: str = str(mydict['key'])

anyway this is more performant than code/reencode cycle just to fool the type checker 😄

Reply:
Let me try..

Reply:
Working, had to take out HA_NOTIFY from the function vector though due to difference in the payload hint from rest of the functions.